### ISSUE TITLE
proc_open() does not accept null in 4th parameter

### ISSUE THREAD

**Issue by gfloyd** [facebook/hhvm#6885]
With `hhvm.php7.scalar_types = 1`, the following code produces a warning:

```
<?php
$process = proc_open('/bin/false', [], $pipes, null);
```

```
Warning: proc_open() expects parameter 4 to be string, null given
```

The docs list `null` as acceptable here: http://php.net/manual/en/function.proc-open.php
It does not produce a warning in php5 or 7, or with `hhvm.php7.scalar_types = 0`.

Note that this currently breaks composer. :)

I am on the nightly release, and also experienced the same in 3.12.

```
$ hhvm --version
HipHop VM 3.13.0-dev (rel)
Compiler: heads/master-0-ga84bdec097fe5adb40587b8a48142d361e4e5911
Repo schema: d3db0494dfde127827da2eb2b308a3fdce8cda83
```

**Comment by Orvid** [comment]
Should just be a matter of updating the function signature to account for possible null values.

**Comment by satyriasizz** [comment]
Hi, It may be bounded to the issue I've just stumbled:

I have HHVM 12.1 using it with nginx on Ubuntu 15.10.
I'm using laravel for my project, so when I tried to run 
`$ php artisan compile`
I got this:
`
root@Peace-EU:/var/www/peacerevolution.net# php artisan compile


Fatal error: Uncaught exception 'ErrorException' with message 'proc_open() expects parameter 4 to be string, null given' in /var/www/peacerevolution.net/vendor/symfony/console/Application.php:898
Stack trace:
#0 (): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(898): proc_open()
#2 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(701): Symfony\Component\Console\Application->getSttyColumns()
#3 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(662): Symfony\Component\Console\Application->getTerminalDimensions()
#4 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(595): Symfony\Component\Console\Application->getTerminalWidth()
#5 /var/www/peacerevolution.net/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(141): Symfony\Component\Console\Application->renderException()
#6 /var/www/peacerevolution.net/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(262): Illuminate\Foundation\Exceptions\Handler->renderForConsole()
#7 /var/www/peacerevolution.net/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(111): Illuminate\Foundation\Console\Kernel->renderException()
#8 /var/www/peacerevolution.net/artisan(36): Illuminate\Foundation\Console\Kernel->handle()
#9 {main}

Exception handler threw an object exception: exception 'ErrorException' with message 'proc_open() expects parameter 4 to be string, null given' in /var/www/peacerevolution.net/vendor/symfony/console/Application.php:898
Stack trace:
#0 (): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(898): proc_open()
#2 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(701): Symfony\Component\Console\Application->getSttyColumns()
#3 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(662): Symfony\Component\Console\Application->getTerminalDimensions()
#4 /var/www/peacerevolution.net/vendor/symfony/console/Application.php(595): Symfony\Component\Console\Application->getTerminalWidth()
#5 /var/www/peacerevolution.net/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(141): Symfony\Component\Console\Application->renderException()
#6 /var/www/peacerevolution.net/bootstrap/cache/compiled.php(2134): Illuminate\Foundation\Exceptions\Handler->renderForConsole()
#7 /var/www/peacerevolution.net/bootstrap/cache/compiled.php(2127): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole()
#8 (): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException()
#9 {main}
`

However, I've put this line:
`hhvm.php7.scalar_types = 0` 
to php.ini and now it's work.

So for me it wasn't just a warning but the error. MB it's coz of 12.1, which I have to use coz 13 handle ~10 requests and fall on my instance.

**Comment by apeabody** [comment]
Looks like this was resolved in #6929, but the correct keyword wasn't included for hhvm-bot to auto close this issue.
